cachito/workers/celery_logging.py

```diff
@@ -54,58 +56,104 @@ def _get_function_arg_value(arg_name, func, args, kwargs):
-    the ``cachito_task_log_format`` config.
+    :param str task_id: the task ID
+    :param class task: the class of the task being executed
+    """
+    logger = logging.getLogger()
+    for handler in logger.handlers:
+        if isinstance(handler, logging.FileHandler):
+            logger.removeHandler(handler)
+
+
+def cleanup_task_logging_customization(task_id, task, **kwargs):
+    """
+    Clean up any logging customizations that were set in ``setup_task_logging_customization``.
 
     :param str task_id: the task ID
     :param class task: the class of the task being executed
     """
     conf = get_worker_config()
 
-    request_id = _get_function_arg_value("request_id", task, kwargs["args"], kwargs["kwargs"])
-    log_filter = AddRequestIDFilter(str(request_id) or "unknown")
     root_logger = logging.getLogger()
     for handler in root_logger.handlers:
         if not isinstance(handler, logging.StreamHandler):
             continue
 
-        handler.addFilter(log_filter)
         if isinstance(handler.formatter, ColorFormatter):
             formatter = ColorFormatter(
-                conf.cachito_task_log_format, use_color=handler.formatter.use_color
+                conf.worker_log_format, use_color=handler.formatter.use_color
             )
         else:
-            formatter = logging.Formatter(conf.cachito_task_log_format)
+            formatter = logging.Formatter(conf.worker_log_format)
         handler.setFormatter(formatter)
 
+        for log_filter in handler.filters:
+            if isinstance(log_filter, AddRequestIDFilter):
+                handler.removeFilter(log_filter)
 
-def cleanup_task_logging(task_id, task, **kwargs):
+
+def setup_task_logging(task_id, task, **kwargs):
+    """
+    Set up the logging for the task via adding a file long handler.
+
+    If ``cachito_request_file_logs_dir`` is set, a temporary log handler is added before the
+    task is invoked.
+    If ``cahito_request_file_logs_dir`` is not set, the temporary log handler will not be added.
+
+    :param str task_id: the task ID
+    :param class task: the class of the task being executed
+    """
+    worker_config = get_worker_config()
+    log_dir = worker_config.cachito_request_file_logs_dir
+    log_level = worker_config.cachito_request_file_logs_level
+    log_format = worker_config.cachito_request_file_logs_format
+
+    request_log_handler = None
+    if log_dir:
+        log_formatter = logging.Formatter(log_format)
+        request_id = _get_function_arg_value(
+            "request_id", task.__wrapped__, kwargs["args"], kwargs["kwargs"]
+        )
+        if not request_id:
+            raise CachitoError("Unable to get 'request_id'")
+
+        log_file_path = os.path.join(log_dir, f"{request_id}.log")
+        request_log_handler = logging.FileHandler(log_file_path)
+        request_log_handler.setLevel(log_level)
+        request_log_handler.setFormatter(log_formatter)
+        os.chmod(log_file_path, 0o666)
```

Comment:
As mentioned in my previous comment, could you make this configurable and have a more secure default? If Cachito were deployed on a single VM for instance, I wouldn't want every single user to be able to tamper with the logs.

Reply:
What should be the value of more secure default?

Reply:
I think `0o660` is a good default.